### PR TITLE
cleanup todo after feature.BlockVolume gate was removed

### PR DESCRIPTION
**Pull request by carlory**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

cleanup todo after feature.BlockVolume gate was removed

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by carlory** [comment]
fix failling test due to the pvc and pv missing volume mode

File: pkg/kubelet/volumemanager/populator/desired_state_of_world_populator_test.go [hunk 1]
```diff
@@ -848,13 +849,13 @@ func TestCreateVolumeSpec_Valid_Nil_VolumeMounts(t *testing.T) {
 		},
 		Spec: v1.PersistentVolumeSpec{
 			ClaimRef:   &v1.ObjectReference{Namespace: "ns", Name: "file-bound"},
-			VolumeMode: nil,
+			VolumeMode: ptr.To(v1.PersistentVolumeFilesystem),
```
Comment (jsafrane) [inline pkg/kubelet/volumemanager/populator/desired_state_of_world_populator_test.go:852]: Is this test change related to the code above?

Comment (carlory) [reply pkg/kubelet/volumemanager/populator/desired_state_of_world_populator_test.go:852]: @jsafrane yes, if the volumeMode is nil, the util.GetVolumeMode func will throw an error and some tests will fail due to the missing volumeMode.
